core/src/main/java/io/github/thibaultbee/streampack/internal/encoders/MediaCodecEncoder.kt

```diff
@@ -220,11 +226,17 @@ abstract class MediaCodecEncoder<T : Config>(
     override fun stopStream() {
         try {
             synchronized(lock) {
+                if(isStopped) {
+                    return
+                }
                 isStopped = true
-                mediaCodec?.setCallback(null)
-                mediaCodec?.signalEndOfInputStream()
+                if(hasInputSurface) {
+                    mediaCodec?.signalEndOfInputStream()
+                }
                 mediaCodec?.flush()
                 mediaCodec?.stop()
+                mediaCodec?.setCallback(null)
+                releaseHandler() // prevent thread leak
```

Comment:
per offline discussion, this code remedies a bug where stopStream always results in mediaCodec throwing an exception, in both setCallback (must be stopped before call) or in signalEndOfStream (not allowed for non-surface encoder)